app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/repositories/MealRepository.java

```diff
@@ -27,13 +35,37 @@ public Task<Meal> getMealById(String id){
             return null;
         });
     }
+
+    /** sends a request to post a meal
+     * @param meal the meal to post
+     * @return the task that may complete, fails if the argument is null, or if the database is unreachable
+     */
     public Task<Void> postMeal(Meal meal){
         if (meal == null) {
             return Tasks.forException(new IllegalArgumentException("Cannot post a null meal"));
         }
+        //We first post the meal to the database,
         return DatabaseFactory.getDependency().add(mealsDataCollectionPath,meal)
                 .continueWithTask(task ->
+                        // and once that is done (and we get the corresponding id of the meal),
                         DatabaseFactory.getDependency().
+
+                                // we need to update the mealId field stored in the database
                                 set(mealsDataCollectionPath,task.getResult(),meal.copyWithId(task.getResult())));
     }
+
+    /** Fetches all the meals stored in the database
+     * @return the task that may complete and contains the meals
+     */
+    public Task<List<Meal>> getAllMeals(){
+        return DatabaseFactory.getDependency().fetchAll(mealsDataCollectionPath).continueWith(t->{
+            ArrayList<Meal> res = new ArrayList<>();
+            if(t.isSuccessful()){
+                for (DocumentSnapshot m: t.getResult().getDocuments()) {
+                    res.add(m.toModel(Meal.class) );
+                }
+            }
+            return res;
+        });
```

Comment:
We should abstract more our repos as I find that we are kinda having similar code to fetch different attributes or sometimes the same attribute in different repos and we can look into fixing that by having more abstraction